### ISSUE TITLE
Maximum call stack size exceeded

### ISSUE THREAD

**Issue by tony19** [ampproject/remapping#163]
Webpack-based projects (including [Vue CLI](https://cli.vuejs.org/) scaffolded projects, and [Quasar](https://quasar.dev/) projects) are failing with `Maximum call stack size exceeded`, starting from `buildSourceMapTree()` and [infinitely recursing in `build()`](https://github.com/ampproject/remapping/blob/9e1ccd9/src/build-source-map-tree.ts#L80). This seems to occur when building the source map.

## Steps to reproduce:

1. Scaffold a new Vue CLI project with:

    ```shell
     npx @vue/cli create --default my-vue-app
    ```

2. Change into the source directory with:

    ```shell
    cd my-vue-app
   ```

3. Start the webpack devserver:

    ```shell
    npm run serve
    ```

4. Observe the error in the console.

<details>
<summary>Call stack</summary>
<pre>
<anonymous> (/Users/tony/src/tmp/test-vue-app123/node_modules/@ampproject/remapping/dist/remapping.umd.js:230)
build (/Users/tony/src/tmp/test-vue-app123/node_modules/@ampproject/remapping/dist/remapping.umd.js:206)
<anonymous> (/Users/tony/src/tmp/test-vue-app123/node_modules/@ampproject/remapping/dist/remapping.umd.js:230)
build (/Users/tony/src/tmp/test-vue-app123/node_modules/@ampproject/remapping/dist/remapping.umd.js:206)
<anonymous> (/Users/tony/src/tmp/test-vue-app123/node_modules/@ampproject/remapping/dist/remapping.umd.js:230)
build (/Users/tony/src/tmp/test-vue-app123/node_modules/@ampproject/remapping/dist/remapping.umd.js:206)
<anonymous> (/Users/tony/src/tmp/test-vue-app123/node_modules/@ampproject/remapping/dist/remapping.umd.js:230)
build (/Users/tony/src/tmp/test-vue-app123/node_modules/@ampproject/remapping/dist/remapping.umd.js:206)
<anonymous> (/Users/tony/src/tmp/test-vue-app123/node_modules/@ampproject/remapping/dist/remapping.umd.js:230)
build (/Users/tony/src/tmp/test-vue-app123/node_modules/@ampproject/remapping/dist/remapping.umd.js:206)
<anonymous> (/Users/tony/src/tmp/test-vue-app123/node_modules/@ampproject/remapping/dist/remapping.umd.js:230)
build (/Users/tony/src/tmp/test-vue-app123/node_modules/@ampproject/remapping/dist/remapping.umd.js:206)
buildSourceMapTree (/Users/tony/src/tmp/test-vue-app123/node_modules/@ampproject/remapping/dist/remapping.umd.js:198)
remapping (/Users/tony/src/tmp/test-vue-app123/node_modules/@ampproject/remapping/dist/remapping.umd.js:273)
mergeSourceMap (/Users/tony/src/tmp/test-vue-app123/node_modules/@babel/core/lib/transformation/file/merge-map.js:19)
generateCode (/Users/tony/src/tmp/test-vue-app123/node_modules/@babel/core/lib/transformation/file/generate.js:75)
run (/Users/tony/src/tmp/test-vue-app123/node_modules/@babel/core/lib/transformation/index.js:55)
transform (/Users/tony/src/tmp/test-vue-app123/node_modules/@babel/core/lib/transform.js:25)
step (/Users/tony/src/tmp/test-vue-app123/node_modules/gensync/index.js:261)
<anonymous> (/Users/tony/src/tmp/test-vue-app123/node_modules/gensync/index.js:273)
async.call.result.err.err (/Users/tony/src/tmp/test-vue-app123/node_modules/gensync/index.js:223)
<anonymous> (/Users/tony/src/tmp/test-vue-app123/node_modules/gensync/index.js:189)
<anonymous> (/Users/tony/src/tmp/test-vue-app123/node_modules/@babel/core/lib/gensync-utils/async.js:74)
<anonymous> (/Users/tony/src/tmp/test-vue-app123/node_modules/gensync/index.js:113)
step (/Users/tony/src/tmp/test-vue-app123/node_modules/gensync/index.js:287)
<anonymous> (/Users/tony/src/tmp/test-vue-app123/node_modules/gensync/index.js:273)
async.call.result.err.err (/Users/tony/src/tmp/test-vue-app123/node_modules/gensync/index.js:223)
</pre>
</details>

## Environment

 - `@ampproject/remapping` - `2.1.1`
 - macOS Big Sur
 - Node 17

**Comment by tony19** [comment]
A workaround is to configure Webpack's [`devtool`](https://webpack.js.org/configuration/devtool/) to something other than `eval` or `eval-source-map` (e.g., you can pick `eval-cheap-source-map`):

*Vue CLI 5.x:*

```js
// vue.config.js
const { defineConfig } = require('@vue/cli-service')
module.exports = defineConfig({
  configureWebpack: {
    devtool: 'eval-cheap-source-map',
  },
})
```

*Vue CLI 4.x:*

```js
// vue.config.js
module.exports = {
  configureWebpack: {
    devtool: 'eval-cheap-source-map',
  },
}
```

**Comment by jridgewell** [comment]
Hi @tony19: This is fixed by https://github.com/babel/babel/pull/14274, and released in Babel v7.17.4. Sorry for the breakage!
